src/com/lilithsthrone/game/character/CharacterUtils.java

```diff
@@ -928,7 +928,7 @@ public static void randomiseBody(GameCharacter character) {
 		}
 		
 		//Ass:
-		if(character.hasFetish(Fetish.FETISH_ANAL_RECEIVING) || character.getHistory()==History.PROSTITUTE) {
```

Comment:
I added in the prostitute background check, as I assumed that the NPC would have lost their anal virginity due to their line of work. The same is for the two cases below (prostitutes would have lost their vaginal virginity, and wouldn't have a pure virgin fetish).

Reply:
I probably should have read through all of your changes before commenting... It makes sense now... Sorry! x_x